commit 168056b072dac39a8b54fec8ee0a1373ce168756 (tag: v5.1.1, refs/changes/55/2255/3)
Author: David Pacheco <dap@joyent.com>
Date:   2017-12-12T15:49:13-08:00 (1 year, 10 months ago)
    
    joyent/node-manta#315 document mlogin session termination conditions
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>
    Approved by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 15b9d7d..179698b 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,10 @@
 
 ## not yet released
 
+## 5.1.1
+
+- joyent/node-manta#315 document mlogin session termination conditions
+
 ## 5.1.0
 
 Minor version bump due to a backwards-compatible addition to the multipart
diff --git a/docs/man/mlogin.md b/docs/man/mlogin.md
index 58f1b59..8def462 100644
--- a/docs/man/mlogin.md
+++ b/docs/man/mlogin.md
@@ -24,6 +24,13 @@ become part of a workflow using interactive terminal utilities on large Manta
 objects without the need to download or transfer the data -- e.g. the the use
 of `mdb` (the Modular Debugger) on crash dumps and core files.
 
+The mlogin session terminates when the top level process has exited (usually the
+interactive shell, unless the `-c` option was used) and all references to that
+process's controlling terminal have also been closed.  Beware that if you fork a
+background process and then exit the shell, the background process (and the job)
+may continue running if they hold the terminal open.  You may need to cancel the
+job in order to stop it.
+
 Note that `mlogin` makes use of publicly-readable assets stored under the
 "poseidon" account, which is provided by the system.
 
diff --git a/man/man1/mlogin.1 b/man/man1/mlogin.1
index 1d26246..c1f4dae 100644
--- a/man/man1/mlogin.1
+++ b/man/man1/mlogin.1
@@ -17,6 +17,13 @@ become part of a workflow using interactive terminal utilities on large Manta
 objects without the need to download or transfer the data \-\- e.g. the the use
 of \fB\fCmdb\fR (the Modular Debugger) on crash dumps and core files.
 .PP
+The mlogin session terminates when the top level process has exited (usually the
+interactive shell, unless the \fB\fC\-c\fR option was used) and all references to that
+process's controlling terminal have also been closed.  Beware that if you fork a
+background process and then exit the shell, the background process (and the job)
+may continue running if they hold the terminal open.  You may need to cancel the
+job in order to stop it.
+.PP
 Note that \fB\fCmlogin\fR makes use of publicly\-readable assets stored under the
 "poseidon" account, which is provided by the system.
 .SH EXAMPLES
@@ -153,4 +160,5 @@ where \fB\fC:login\fR is the account login name.
 .PP
 DSA keys do not work when loaded via the SSH agent.
 .PP
-Report bugs at Github \[la]https://github.com/joyent/node-manta/issues\[ra]
+Report bugs at Github
+\[la]https://github.com/joyent/node-manta/issues\[ra]
diff --git a/package.json b/package.json
index d819c40..9dd62ec 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
         "type": "git",
         "url": "git://github.com/joyent/node-manta.git"
     },
-    "version": "5.1.0",
+    "version": "5.1.1",
     "main": "./lib/index.js",
     "dependencies": {
         "assert-plus": "^1.0.0",
